eng/Version-Increment.ps1

```diff
@@ -0,0 +1,59 @@
+<#
+.SYNOPSIS
+Bumps up package versions after release
+
+.DESCRIPTION
+This script bumps up package versions following conventions defined at https://github.com/Azure/azure-sdk/blob/master/docs/policies/releases.md#incrementing-after-release-net
+
+.PARAMETER PackageCsprojPath
+Link to the package csproj
+
+.EXAMPLE
+Updating package version for core
+
+Version-Increment.ps1 -PackageCsprojPath "C:\Git\azure-sdk-for-net\sdk\core\Azure.Core\src\Azure.Core.csproj"
+
+#>
+
+[CmdletBinding()]
+Param (
+    [Parameter(Position=0)]
+    [ValidateNotNullOrEmpty()]
+    [string] $PackageCsprojPath
+)
+
+$VERSION_REGEX = "^(?<major>\d+)(\.(?<minor>\d+))?(\.(?<patch>\d+))?(\-(?<pre>[0-9A-Za-z\-\.]+))?(\+(?<build>[0-9A-Za-z\-\.]+))?$"
+$CsprojData = New-Object -TypeName XML
+$CsprojData.Load($PackageCsprojPath)
+
+$PackageVersion = Select-XML -Xml $CsprojData -XPath '/Project/PropertyGroup/Version'
+Write-Verbose "Current Version: ${PackageVersion}"
+
+$PackageVersion -Match $VERSION_REGEX 
+
+if($Matches['pre'] -eq $null){$pre = @()}
+else{$pre = $Matches['pre'].Split(".")}
+
+$major = [int]$Matches['major']
+$minor = [int]$Matches['minor']
+$patch = [int]$Matches['patch']
+$pretag = $pre[0]
+$preVer = [int]$pre[1]
+
+# Increment Version
+if($pretag -eq 'preview')
+{
+    $preVer++
+}
+else
+{
+    $pretag = 'preview'
```

Comment:
We should ensure $pretag isn't set to something else here and if it is we should error. I think I would change this logic to 
```
if ($pretag -eq $null)
{
   $pretag = 'preview'
   $preVer = 1
    $minor++
    $patch = 0
}
elseif ($pretag -eq 'preview' -and $preVer -ne $null)
{
  $preVer++
}
else
{
   Write-Error "Unexpected pre-release identifier $pretag" 
}
```